### PR TITLE
LPS-94126 | Conjunction unit tests

### PR DESCRIPTION
**Pull request by andresfulla**

### CODE REVIEW

**Comment by andresfulla** [comment]
ci:test:sf

**Comment by andresfulla** [comment]
ci:test:relevant

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
ci:test:relevant

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js [hunk 1]
```diff
@@ -38,6 +38,15 @@ function Conjunction({
...
+	const [activeLabel, setActiveLabel] = useState(null);
+	useEffect(() => {
+		const selectedConjunction = supportedConjunctions.find(
+			c => c.name === conjunctionName
+		);
+
+		setActiveLabel(selectedConjunction.label);
+	}, [conjunctionName, supportedConjunctions]);
+
 	function _handleItemClick(conjuntionName) {
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js]: conjuntion means nothing, conjunction!!!!

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		const [selectedConjunction] = getAllByText(initialConjunctionLabel);
+
+		expect(dropdownMenu.className).not.toContain('show');
+
+		userEvent.click(selectedConjunction);
+
+		expect(dropdownMenu.className).toContain('show');
+	});
+
+	it('onSelect prop is triggered on selection with conjuntionName param in editable mode', () => {
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js:109]: conjuntion again...

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js (hunk 1 above)
Comment (dgarciasarai) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js]: 😅

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js [hunk 3]
```diff
@@ -38,6 +38,15 @@ function Conjunction({
 		className
 	);
 
+	const [activeLabel, setActiveLabel] = useState(null);
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js:41]: good naming? means `activeConjunction`?

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+				supportedConjunctions={conjunctions}
+			/>,
+			{
+				baseElement: document.body
+			}
+		);
+
+		const [selectedConjunction] = getAllByText(initialConjunctionLabel);
+
+		expect(selectedConjunction.tagName).toBe('BUTTON');
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js:55]: check label and name here?

Comment (dgarciasarai) [reply modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js:55]: check if the button is not disabled?

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+				supportedConjunctions={conjunctions}
+			/>,
+			{
+				baseElement: document.body
+			}
+		);
+
+		const selectedConjunction = getByText(initialConjunctionLabel);
+
+		expect(selectedConjunction.tagName).toBe('DIV');
```
Comment (dgarciasarai) [inline modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js:78]: button when editable and div when not editable, right?

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js (hunk 3 above)
Comment (andresfulla) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/Conjunction.es.js:41]: is the `activeConjunctionLabel`, but given we're in the `Conjunction` component and is an internal state variable... I'll call it an `activeLabel`

File: modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js (hunk 4 above)
Comment (andresfulla) [reply modules/apps/segments/segments-web/test/js/components/criteria_builder/Conjunction.es.js:55]: button is never disabled.
name is not present in the UI.

**Comment by andresfulla** [comment]
just squash commits

**Pull request closed without merging**